jre_emul/android/platform/libcore/ojluni/src/main/java/java/util/jar/Pack200.java

```diff
@@ -0,0 +1,825 @@
+         * with <tt>com.</tt> or a similar prefix.
+         * All property names beginning with <tt>pack.</tt> and
+         * <tt>unpack.</tt> are reserved for use by this API.
+         * <p>
+         * Unknown properties may be ignored or rejected with an
+         * unspecified error, and invalid entries may cause an
+         * unspecified error to be thrown.
+         *
+         * <p>
+         * The returned map implements all optional {@link SortedMap} operations
+         * @return A sorted association of property key strings to property
+         * values.
+         */
+        SortedMap<String,String> properties();
+
+        /**
+         * Takes a JarFile and converts it into a Pack200 archive.
+         * <p>
+         * Closes its input but not its output.  (Pack200 archives are appendable.)
+         * @param in a JarFile
+         * @param out an OutputStream
+         * @exception IOException if an error is encountered.
+         */
+        void pack(JarFile in, OutputStream out) throws IOException ;
+
+        /**
+         * Takes a JarInputStream and converts it into a Pack200 archive.
+         * <p>
+         * Closes its input but not its output.  (Pack200 archives are appendable.)
+         * <p>
+         * The modification time and deflation hint attributes are not available,
+         * for the JAR manifest file and its containing directory.
+         *
+         * @see #MODIFICATION_TIME
+         * @see #DEFLATE_HINT
+         * @param in a JarInputStream
+         * @param out an OutputStream
+         * @exception IOException if an error is encountered.
+         */
+        void pack(JarInputStream in, OutputStream out) throws IOException ;
+
+        /**
+         * Registers a listener for PropertyChange events on the properties map.
+         * This is typically used by applications to update a progress bar.
+         *
+         * <p> The default implementation of this method does nothing and has
+         * no side-effects.</p>
+         *
+         * <p><b>WARNING:</b> This method is omitted from the interface
+         * declaration in all subset Profiles of Java SE that do not include
+         * the {@code java.beans} package. </p>
+
+         * @see #properties
+         * @see #PROGRESS
+         * @param listener  An object to be invoked when a property is changed.
+         * @deprecated The dependency on {@code PropertyChangeListener} creates
+         *             a significant impediment to future modularization of the
+         *             Java platform. This method will be removed in a future
+         *             release.
+         *             Applications that need to monitor progress of the packer
+         *             can poll the value of the {@link #PROGRESS PROGRESS}
+         *             property instead.
+         */
+        @Deprecated
+        default void addPropertyChangeListener(PropertyChangeListener listener) {
+        }
```

Comment:
Please comment out the methods that support PropertyChangeListener and its import statement. Besides the valid comments for deprecating them, it also adds a jre_beans_lib dependency on apps using the jre_zip_lib JRE subset, increasing app size unnecessarily.

Reply:
I see. Just commented out everything about PropertyChangeListener.

Reply:
LGTM